debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java

```diff
@@ -156,9 +156,8 @@ private String tableSchemaName(TableId tableId) {
         else if (Strings.isNullOrEmpty(tableId.schema())) {
             return tableId.catalog() + "." + tableId.table();
         }
-        // When both catalog and schema is present then only schema is used
         else {
-            return tableId.schema() + "." + tableId.table();
+            return tableId.catalog() + "." + tableId.schema() + "." + tableId.table();
```

Comment:
You mean the connectors that do support/populate the catalog name but don't currently use it to build the destination topic name?

Would it be acceptable to introduce a configuration parameter that would allow to opt-in and then opt out of the new naming schema? Once agreed on the migration path, we can implement it outside of this branch.